torch/csrc/generic/Tensor.cpp

```diff
@@ -548,15 +548,26 @@ static bool THPTensor_(_checkAdvancedIndexing)(THPTensor *indexed, PyObject *arg
   // adjacent
 
   long ndim = THTensor_(nDimension)(LIBRARY_STATE indexed->cdata);
-  if (PySequence_Check(arg) && PySequence_Size(arg) <= ndim) {
+
+  if (PySequence_Check(arg) && PySequence_Size(arg) <= ndim + 1) {
     THPObjectPtr fast = THPObjectPtr(PySequence_Fast(arg, NULL));
 
     bool sequenceFound = false;
     bool nonColonEllipsisFound = false;
     bool ellipsisFound = false;
     Py_ssize_t lastSeqDim = -1;
 
+    // Note that we can have ndim + 1 Tensors in the case where we have an ellipsis,
```

Comment:
if ndim + 1 Tensors are given, but no ellipsis is found, i dont see where it cuts to go give an error message. Is that missing? Can you add a test for that case (which asserts that an error is given)